### ISSUE TITLE
Support for streaming local files via AirPlay

### ISSUE THREAD

**Issue by postlund** [postlund/pyatv#95]
I started this work a while back and I think I have a branch for it somewhere. Basically setup a local web server serving the file and pass URL to the Apple TV. Should be simple enough.

**Comment by postlund** [comment]
I had some issues with the aiohttp web server when I initially started working on this, but it seems to work as expected now. Used this as a sample web server:

```python
import pathlib

from aiohttp import web
from aiohttp.web import middleware

@middleware
async def middleware(request, handler):
    if str(request.rel_url) == "/video.mp4":
        return await handler(request)
    return web.Response(body="Permission denied", status=401)

app = web.Application(middlewares=[middleware])

app.router.add_static('/', pathlib.Path(__file__).parent, show_index=True)

web.run_app(app)
```

Serving the Big Buck Bunny video (http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4) as example by streaming like this:

```shell
$ atvremote --debug -s 10.0.10.101 play_url=http://10.0.10.254:8080/video.mp4
```

This worked as expected. IP .101 is my Apple TV (3rd gen in this case) and .254 is the server running the code above. There will be some tinkering to get this in place, but shouldn't be that hard.

**Comment by postlund** [comment]
Managed to extend the example and integrate it into `pyatv` (hack-ish). I changed `play_url` to check if the given URL is a file available on the system and if that's the case serve it via a temporary web server and then stream from that. Worked on the first try. Simply amazing!

This is proof-of-concept at the moment but looks very promising so far. Not sure when I will finish it but if possible I will continue working on it tomorrow. Testing is something I will have to think about since it seems quite tedious to mock files.
